zip-bolt-support.rst

```diff
@@ -0,0 +1,272 @@
+* ``vShieldedSpend[0]``: tx for customer’s note commitment and nullifier for the coins
+  
+  - ``cv``: commitment for the input note
+  - ``root``: root hash of note commitment tree at some block height
+  - ``nullifier``: unique serial number of the input note
+  - ``rk``: randomized pubkey for spendAuthSig
+  - ``zkproof``: zero-knowledge proof for the note
+  - ``spendAuthSig``: signature authorizing the spend
+* ``vShieldedSpend[1]``: tx for merchant’s note commitment and nullifier for the coins (if dual-funded)
+  
+  - ``cv``: commitment for the input note
+  - ``root``: root hash of note commitment tree at some block height
+  - ``nullifier``: unique serial number of the input note
+  - ``rk``: randomized pubkey for spendAuthSig
+  - ``zkproof``: zero-knowledge proof for the note
+  - ``spendAuthSig``: signature authorizing the spend
+* ``tx_out_count``: 1
+* ``tx_out``: (using a P2SH output)
+
+  - ``scriptPubKey`` must have the form ``0 <32-byte hash>``, where the latter is the hash of the script needed to spend the output.
+To redeem this output, the redeeming transaction must present:
+
+	scriptSig: 0 0 <channel-token> <channel-pubkey> <cust-sig> <merch-sig> <serializedScript>, 
+	
+where ``serializedScript`` is as follows: 
+	
+	2 <cust-pubkey> <merchant-pubkey> 2 OP_CHECKMULTISIGVERIFY 
+	OP_DUP OP_HASH160 <hash-of-channel-token> OP_EQUALVERIFY OP_BOLT
+
+* ``bindingSig``: a signature that proves that (1) the total value spent by Spend transfers - Output transfers = value balance field.
+
+[TODO: provide example transaction here.]
+
+2.3 Initial Wallet Commitment
+-------------
+The initial commitment transaction is generated by the customer during the channel initialization but does not broadcast to the network.
+
+* ``version``: specify version number
+* ``groupid``: specify group id
+* ``locktime``: should be set such that commitment transactions can be included in a current block.
+* ``txin`` count: 1
+
+- ``txin[0]`` outpoint: references the funding transaction txid and output_index
+- ``txin[0]`` script bytes: 0
+- ``txin[0]`` witness: 0 <channel-token> <wallet-pubkey> <cust-sig> <merch-sig> <2 <cust_fund_pubkey> <merch_fund_pubkey> 2 OP_CHECKMULTISIGVERIFY OP_DUP OP_HASH160 <hash-of-channel-token> OP_EQUALVERIFY OP_BOLT>
+* nShieldedOutput: 2
+
+- ``vShieldedOutput[0-1]``: 
+- ``cv``: commitment for the input note
+- ``cmu``:...
+- ``ephemeralKey``:ephemeral public key
+- ``encCiphertext``: encrypted output note (part 1)
+- ``outCiphertext``: encrypted output note (part 2)
+- ``zkproof``: zero-knowledge proof for the note
+
+[TODO: update commitment transaction to reflect transaction in reference implementation.]
+
+2.4 Channel Closing
+-------------
+To close the channel, the customer can initiate by posting a commitment transaction that spends from the multi-signature transparent address with a witness that satisfies the script and the ``OP_BOLT`` opcode.
```

Comment:
Better: "To initiate channel closing, the customer can post a commitment transaction"